client/asset/dcr/dcr.go

```diff
@@ -436,31 +455,55 @@ func (dcr *ExchangeWallet) fund(confs uint32,
 		return nil
 	}
 
-out:
-	for {
-		// If there are none left, we don't have enough.
-		if len(utxos) == 0 {
-			return nil, 0, 0, fmt.Errorf("not enough to cover requested funds")
+	tryUTXOs := func(minconf int64) (ok bool, err error) {
+		sum, size = 0, 0
+		coins, spents = nil, nil
+
+		okUTXOs := make([]*compositeUTXO, 0, len(utxos)) // over-allocate
+		for _, cu := range utxos {
+			if cu.confs >= minconf {
+				okUTXOs = append(okUTXOs, cu)
+			}
 		}
-		// On each loop, find the smallest UTXO that is enough. If no UTXO is large
-		// enough, add the largest and continue.
-		var txout *compositeUTXO
-		for _, txout = range utxos {
-			if enough(sum, size, txout) {
-				err = addUTXO(txout)
-				if err != nil {
-					return nil, 0, 0, err
+
+		for {
+			// If there are none left, we don't have enough.
+			if len(okUTXOs) == 0 {
+				return false, nil
+			}
+			// On each loop, find the smallest UTXO that is enough.
+			for _, txout := range okUTXOs {
+				if enough(sum, size, txout) {
+					if err = addUTXO(txout); err != nil {
+						return false, err
+					}
+					return true, nil
 				}
-				break out
 			}
+			// No single UTXO was large enough. Add the largest (the last
+			// output) and continue.
+			if err = addUTXO(okUTXOs[len(okUTXOs)-1]); err != nil {
+				return false, err
+			}
+			// Pop the utxo.
+			okUTXOs = okUTXOs[:len(okUTXOs)-1]
 		}
-		// Append the last output, which is the largest.
-		err = addUTXO(txout)
+	}
+
+	// First try with confs>0.
+	ok, err := tryUTXOs(1)
+	if err != nil {
+		return nil, 0, 0, err
+	}
+	// Fallback to allowing 0-conf outputs.
+	if !ok {
+		ok, err = tryUTXOs(0)
```

Comment:
The least we can do until we have safe/trusted support with dcrwallet: https://github.com/decred/dcrwallet/issues/1769